### ISSUE TITLE
Build failure on RHEL 7.4 beta

### ISSUE THREAD

**Issue by bmerchant** [zfsonlinux/zfs#6234]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | RHEL Server
Distribution Version    | 7.4 beta
Linux Kernel                 | 3.10.0-663.el7.x86_64
Architecture                 | x86_64
ZFS Version                  | 0.6.5.9-1, 0.7.0-rc4, 0.7.0-rc4_62_gef4be34
SPL Version                  | 0.6.5.9-1, 0.7.0-rc4, 0.7.0-rc4_4_gac48361

### Describe the problem you're observing

ZFS build failure on RHEL 7.4 beta

### Describe how to reproduce the problem

Try to build ZFS on RHEL 7.4 beta

### Include any warning/errors/backtraces from the system logs

Kernel include/linux/bio.h now includes a macro for bio_set_op_attrs that conflicts with the ifndef in ZFS build/include/linux/blkdev_compat.h starting at line 322. For now, I'm deleting the ifndef statement entirely as a workaround.

```
/var/lib/dkms/zfs/0.7.0/build/include/linux/blkdev_compat.h:322:18: error: expected identifier or '(' before 'struct'
 bio_set_op_attrs(struct bio *bio, unsigned rw, unsigned flags)
                  ^
include/linux/bio.h:48:44: note: in definition of macro 'bio_set_op_attrs'
 #define bio_set_op_attrs(bio, op, flags) ((bio)->bi_rw |= (op | flags))
                                            ^
include/linux/bio.h:48:48: error: expected ')' before '->' token
 #define bio_set_op_attrs(bio, op, flags) ((bio)->bi_rw |= (op | flags))
                                                ^
/var/lib/dkms/zfs/0.7.0/build/include/linux/blkdev_compat.h:322:1: note: in expansion of macro 'bio_set_op_attrs'
 bio_set_op_attrs(struct bio *bio, unsigned rw, unsigned flags)
```

```

**Comment by tonyhutter** [comment]
This fixes it for me on RHEL 7.4:
```diff --git a/config/kernel-bio-op.m4 b/config/kernel-bio-op.m4
index 5559d6c..8299e49 100644
--- a/config/kernel-bio-op.m4
+++ b/config/kernel-bio-op.m4
@@ -69,7 +69,7 @@ AC_DEFUN([ZFS_AC_KERNEL_BIO_BI_OPF], [
 AC_DEFUN([ZFS_AC_KERNEL_HAVE_BIO_SET_OP_ATTRS], [
        AC_MSG_CHECKING([whether bio_set_op_attrs is available])
        ZFS_LINUX_TRY_COMPILE([
-               #include <linux/blk_types.h>
+               #include <linux/bio.h>
        ],[
                struct bio *bio __attribute__ ((unused)) = NULL;
 ```
My guess is that this will break other builds though.  Let me look into it...

**Comment by tonyhutter** [comment]
Fix out for review: https://github.com/zfsonlinux/zfs/pull/6271
